File: java-example/src/main/java/org/onflow/examples/java/get_block/GetBlockAccessAPIConnector.java
package org.onflow.examples.java.get_block;

import org.onflow.flow.sdk.*;

public class GetBlockAccessAPIConnector {

    private final FlowAccessApi accessAPI;

    public GetBlockAccessAPIConnector(FlowAccessApi accessAPI) {
        this.accessAPI = accessAPI;
    }

    public FlowBlock getLatestSealedBlock() {
        boolean isSealed = true;
        FlowAccessApi.AccessApiCallResponse<FlowBlock> response = accessAPI.getLatestBlock(isSealed);

        if (response instanceof FlowAccessApi.AccessApiCallResponse.Success) {
            return ((FlowAccessApi.AccessApiCallResponse.Success<FlowBlock>) response).getData();
        } else {
            FlowAccessApi.AccessApiCallResponse.Error errorResponse = (FlowAccessApi.AccessApiCallResponse.Error) response;
            throw new RuntimeException(errorResponse.getMessage(), errorResponse.getThrowable());
        }
    }

    public FlowBlock getBlockByID(FlowId blockID) {
        FlowAccessApi.AccessApiCallResponse<FlowBlock> response = accessAPI.getBlockById(blockID);

        if (response instanceof FlowAccessApi.AccessApiCallResponse.Success) {
            return ((FlowAccessApi.AccessApiCallResponse.Success<FlowBlock>) response).getData();
        } else {
            FlowAccessApi.AccessApiCallResponse.Error errorResponse = (FlowAccessApi.AccessApiCallResponse.Error) response;
            throw new RuntimeException(errorResponse.getMessage(), errorResponse.getThrowable());
        }
    }

    public FlowBlock getBlockByHeight(long height) {
        FlowAccessApi.AccessApiCallResponse<FlowBlock> response = accessAPI.getBlockByHeight(height);

        if (response instanceof FlowAccessApi.AccessApiCallResponse.Success) {
            return ((FlowAccessApi.AccessApiCallResponse.Success<FlowBlock>) response).getData();
        } else {
            FlowAccessApi.AccessApiCallResponse.Error errorResponse = (FlowAccessApi.AccessApiCallResponse.Error) response;
            throw new RuntimeException(errorResponse.getMessage(), errorResponse.getThrowable());
        }
    }
}

File: kotlin-example/src/main/kotlin/org/onflow/examples/kotlin/get_block/GetBlockAccessAPIConnector.kt
package org.onflow.examples.kotlin.get_block

import org.onflow.flow.sdk.*

internal class GetBlockAccessAPIConnector(
    private val accessAPI: FlowAccessApi
) {
    fun getLatestSealedBlock(): FlowBlock {
        val isSealed = true
        return when (val response = accessAPI.getLatestBlock(isSealed)) {
            is FlowAccessApi.AccessApiCallResponse.Success -> response.data
            is FlowAccessApi.AccessApiCallResponse.Error -> throw Exception(response.message, response.throwable)
        }
    }

    fun getBlockByID(blockID: FlowId): FlowBlock {
        return when (val response = accessAPI.getBlockById(blockID)) {
            is FlowAccessApi.AccessApiCallResponse.Success -> response.data
            is FlowAccessApi.AccessApiCallResponse.Error -> throw Exception(response.message, response.throwable)
        }
    }

    fun getBlockByHeight(height: Long): FlowBlock {
        return when (val response = accessAPI.getBlockByHeight(height)) {
            is FlowAccessApi.AccessApiCallResponse.Success -> response.data
            is FlowAccessApi.AccessApiCallResponse.Error -> throw Exception(response.message, response.throwable)
        }
    }
}
